component/src/main/java/org/ballerinalang/kafka/nativeimpl/consumer/endpoint/RegisterListener.java

```diff
@@ -49,28 +50,34 @@
         orgName = ORG_NAME,
         packageName = PACKAGE_NAME,
         functionName = "registerListener",
-        receiver = @Receiver(type = TypeKind.OBJECT, structType = CONSUMER_ENDPOINT_STRUCT_NAME,
-                structPackage = KAFKA_NATIVE_PACKAGE),
-        args = {@Argument(name = "serviceType", type = TypeKind.TYPEDESC)}
+        receiver = @Receiver(
+                type = TypeKind.OBJECT,
+                structType = CONSUMER_ENDPOINT_STRUCT_NAME,
+                structPackage = KAFKA_NATIVE_PACKAGE
+        )
 )
 public class RegisterListener implements NativeCallableUnit {
 
     @Override
     public void execute(Context context, CallableUnitCallback callableUnitCallback) {
         Service service = BLangConnectorSPIUtil.getServiceRegistered(context);
-        BMap<String, BValue> consumerEndpoint = (BMap<String, BValue>) context.getRefArgument(0);
-        BMap<String, BValue> consumerConfig = (BMap<String, BValue>) consumerEndpoint.get("consumerConfig");
-
-        Properties configParams = KafkaUtils.processKafkaConsumerConfig(consumerConfig);
+        Struct serviceEndpoint = BLangConnectorSPIUtil.getConnectorEndpointStruct(context);
+        BMap<String, BValue> serverConfigs = (BMap<String, BValue>) context.getRefArgument(0);
+        BMap<String, BValue> consumerConfigs = (BMap<String, BValue>) serverConfigs.get("consumerConfig");
```

Comment:
Shall we take consumerConfig to a meaningful constant, since it has been used several places

Reply:
Agreed. Will do.